opm/material/fluidsystems/BlackOilFluidSystem.hpp

```diff
@@ -655,6 +654,74 @@ class BlackOilFluidSystem : public BaseFluidSystem<Scalar, BlackOilFluidSystem<S
         throw std::logic_error("Unhandled phase index "+std::to_string(phaseIdx));
     }
 
+    template <class FluidState, class LhsEval = typename FluidState::Scalar>
+    static LhsEval energy(const FluidState& fluidState,
+                                    unsigned phaseIdx,
+                                    unsigned regionIdx)
+    {
```

Comment:
that function should be named either `internalEnergy()` or `enthalpy()`. Also, `internalEnergy()` and `enthalpy()` need to be consistent, i.e. h_alpha = u_alpha + p_alpha/rho_alpha.

Reply:
ah, okay. maybe this can even replace the ´enthalpy()` and/or `internalEnergy()` functions directly?